### ISSUE TITLE
Error trying to apply an icon in Mario Odyssey

### ISSUE THREAD

**Issue by mxarment** [Slluxx/IconGrabber#5]
"Se ha cerrado el programa debido a un error" The program was closed due to an error

**Comment by Slluxx** [comment]
Not much i can do with that little information. Did you try a different icon?

**Comment by mxarment** [comment]
Yes, every icon make the game crash but only that game, I had no problem with others
I also try to reinstall the game and to try another version of it but nothing works
Just opening an issue here in case you know of this because i think the problem has to be the game itself
![IMG_20230803_012328](https://github.com/Slluxx/IconGrabber/assets/109087512/cb7b73a9-5344-4930-aff4-fcfc8056ce22)

**Comment by Slluxx** [comment]
I will test this later with the game.

**Comment by sodasoba1** [comment]
> 

This a very unusual error, are you using the latest version of sys-tweak

**Comment by mxarment** [comment]
> > 
> 
> This a very unusual error, are you using the latest version of sys-tweak

yes, I think so. I'm fairly new, I hacked the switch and install everything just a week ago but I dont remember how I installed sys-tweak but Im 99% sure is the last one

**Comment by Slluxx** [comment]
> yes, I think so. I'm fairly new, I hacked the switch and install everything just a week ago but I dont remember how I installed sys-tweak but Im 99% sure is the last one

Where did you get it from? sys-tweak is not downloadable from the original repository for months now so wherever you got it from, it very well can be an outdated version.

Attached is a private build i made 2 weeks ago.

Download, extract, rename to exefs.nsp and overwrite yours in `/atmosphere/contents/00FF747765616BFF/`
[sys-tweak.nsp.zip](https://github.com/Slluxx/IconGrabber/files/12267307/sys-tweak.nsp.zip)

**Comment by mxarment** [comment]
> 

thank you for the file. I just updated it and it still gives me the same error :/
Because it is a very specific problem with a game only It doesnt matter if we cant figure out the issue

Thank you for giving me support and for the app :)

**Comment by rlaphoenix** [comment]
Can you try go into Data Management -> Software -> Mario Odyssey -> Check for Corrupted Data and see if that is the reason?

**Comment by mxarment** [comment]
> Can you try go into Data Management -> Software -> Mario Odyssey -> Check for Corrupted Data and see if that is the reason?

I'm sorry I didn't answer sooner, I finally managed to solve it.
For some reason square resolutions other than 256*256 trigger the error, I tried it in another game and the same thing happened.
![Captura de pantalla 2023-08-06 004732](https://github.com/Slluxx/IconGrabber/assets/109087512/91104ec1-6a3b-4d74-9474-621e30d67525)
(Right side is the icon that fails to load, the other ones are the icon of another games I used to compare)

Changing the resolution and with 24 bits of depth I put the icon in the folder of mario oddysey, in atmosphere/contents/0100000000010000 with the name of «icon».
![Captura de pantalla 2023-08-06 011311](https://github.com/Slluxx/IconGrabber/assets/109087512/9ed89fc4-628c-4b66-964a-8e1d4f4e0ffd)

Doing this manual change everything worked perfectly, thank you very much for the help these days.

**Comment by rlaphoenix** [comment]
While this is expected, it's not expected for the icon file downloaded by IconGrabber to be anything other than 256x256. Are you saying IconGrabber downloaded and saved the icon as 400x400?

**Comment by Slluxx** [comment]
@mxarment just like @rlaphoenix said, this is expected behavior and has nothing todo with icongrabber **unless** icongrabber actually did not convert the size.
